Controller/ConnectController.php

```diff
@@ -299,6 +300,14 @@ public function redirectToServiceAction(Request $request, $service)
 
                 $param = $this->container->getParameter('hwi_oauth.target_path_parameter');
                 if (!empty($param) && $targetUrl = $request->get($param)) {
+                    $domainsWhiteList = $this->container->getParameter('hwi_oauth.target_path_domains_whitelist');
+                    if (count($domainsWhiteList) > 0 &&
+                        ($urlParts = parse_url($targetUrl)) &&
+                        !in_array($urlParts['host'], $domainsWhiteList, true)
```

Comment:
To logic could be moved to a service. Then, it would be easy to add a unit test as well.

Reply:
I'm not entirely sure where to put this new service? Any suggestions? :)

Reply:
Something like `HWI\Bundle\OAuthBundle\Util\DomainWhitelist`.